default.nix

```diff
@@ -0,0 +1,10 @@
+{pkgs ? import <nixpkgs> {}} :
+
+pkgs.mkShell {
+  name = "nixos-up";
+  buildInputs = with pkgs; [ ocaml jq ];
+  shellHook = ''
+    ocaml ${./nixos-up.ml}
+    exit $?
```

Comment:
Since we're just running a single command, `mkShell` with `exit` feels a little bit roundabout. Do you know if there's any nix way to package an application and run it in an ad hoc way like this? 

I'm not personally familiar with any alternatives so I'm happy to go the nix-shell route, but it just seems like there ought to be a more direct solution.

Reply:
Not totally sure, but I think you have to do something with a `shellHook` if you want it to integrate with nix-shell, and at that point might as well use mkShell.

Reply:
One con of this `mkShell` approach vs the `#!nix-shell` script is that if you Ctrl-C halfway through the execution of the execution, you get dumped into the nix-shell as opposed to back out into the root shell. Not a big deal, but a little bit annoying.